contrib/examples/kafka/src/main/java/com/google/cloud/dataflow/contrib/kafka/examples/TopHashtagsExample.java

```diff
@@ -0,0 +1,257 @@
+
+    @Description("Number of Top Hashtags to track")
+    @Default.Integer(10)
+    Integer getNumTopHashtags();
+    void setNumTopHashtags(Integer count);
+
+    @Description("Kafka topic name for writing results")
+    @Required
+    String getOutputTopic();
+    void setOutputTopic(String topic);
+  }
+
+  public static void main(String args[]) {
+
+    Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
+
+    Pipeline pipeline = Pipeline.create(options);
+
+    KafkaIO.Read<?, String> reader = KafkaIO.read()
+        .withBootstrapServers(options.getBootstrapServers())
+        .withTopics(options.getTopics())
+        .withValueCoder(StringUtf8Coder.of())
+        .withTimestampFn(timestampFn);
+
+    pipeline
+      .apply("sample_tweets", reader)
+      .apply(Values.<String>create())
+      .apply(ParDo.of(new ExtractHashtagsFn()))
+      .apply(Window.<String>into(SlidingWindows
+          .of(Duration.standardMinutes(options.getSlidingWindowLengthMinutes()))
+          .every(Duration.standardMinutes(options.getSlidingWindowIntervalMinutes()))))
+      .apply(Count.<String>perElement())
+      .apply(Top.of(options.getNumTopHashtags(), new KV.OrderByValue<String, Long>())
+                .withoutDefaults())
+      .apply(ParDo.of(new OutputFormatter()))
+      .apply(ParDo.of(new KafkaWriter(options)));
+
+    pipeline.run();
+  }
+
+  // The rest of the file implements DoFns to extract hashtags, formatting output, writing output
+  // back to Kafka. Note that writing to Kafka is not a complete Dataflow Sink. It is a best-effort
+  // logging of the results.
+
+  private static final ObjectMapper jsonMapper = new ObjectMapper();
+
+  /**
+   * Emit hashtags in the tweet (if any).
+   */
+  private static class ExtractHashtagsFn extends DoFn<String, String> {
+
+    @Override
+    public void processElement(ProcessContext ctx) throws Exception {
+      for (JsonNode hashtag : jsonMapper.readTree(ctx.element())
+          .with("entities")
+          .withArray("hashtags")) {
+        ctx.output(hashtag.get("text").asText());
+      }
+    }
+  }
+
+  // extract timestamp from "timestamp_ms" field.
+  private static SerializableFunction<KV<byte[], String>, Instant> timestampFn =
+      new SerializableFunction<KV<byte[], String>, Instant>() {
+        @Override
+        public Instant apply(KV<byte[], String> kv) {
+          try {
+            long tsMillis = jsonMapper.readTree(kv.getValue()).path("timestamp_ms").asLong();
+            return tsMillis == 0 ? Instant.now() : new Instant(tsMillis);
+          } catch (Exception e) {
+            throw Throwables.propagate(e);
+          }
+        }
+      };
+
+  // return json string containing top hashtags and window information time
+  private static class OutputFormatter extends DoFn<List<KV<String, Long>>, String>
+      implements DoFn.RequiresWindowAccess {
+
+    private transient DateTimeFormatter formatter;
```

Comment:
why not make this `private static final` rather than re-allocating per bundle? `DateTimeFormatter` is threadsafe, right?